docs/sphinxext/pyshinyapp.py

```diff
@@ -46,10 +48,15 @@ def html(self):
             # TODO: allow the layout to be specified (right now I don't think we need
             # horizontal layout, but maybe someday we will)
             code = (
-                "#| standalone: true\n#| components: [editor, viewer]\n#| layout: vertical\n#| viewerHeight: 400\n"
+                "#| standalone: true\n#| components: [editor, viewer]\n#| layout: vertical\n#| viewerHeight: 400\n## file: app.py\n"
                 + code
             )
 
+            for f in self["files"]:
+                with open(f, "rb") as x:
+                    fc = base64.b64encode(x.read()).decode("utf-8")
+                    code += f"\n\n## file: {os.path.basename(f)}\n## type: binary\n{fc}"
```

Comment:
I don't think this should encode all files as binary. Text files shouldn't be base64 encoded -- that will work, but it'll make the file non-editable in the editor, and the content will also not be viewable.

Here's some code I used in shinylive to attempt reading as text, and if that fails, read as binary. It would make sense to do something similar here.
https://github.com/rstudio/py-shinylive/blob/486f9f9f7f54cb129b95c0122716990a40d66ed3/shinylive/_app_json.py#L73-L82